docs/tutorials/go-built-in.md

```diff
@@ -103,579 +101,527 @@ func NewKVStoreApplication() *KVStoreApplication {
+func (app *KVStoreApplication) CheckTx(req abcitypes.RequestCheckTx) abcitypes.ResponseCheckTx {
  return abcitypes.ResponseCheckTx{Code: 0}
 }
 
-func (KVStoreApplication) Commit() abcitypes.ResponseCommit {
+func (app *KVStoreApplication) Commit() abcitypes.ResponseCommit {
  return abcitypes.ResponseCommit{}
 }
 
-func (KVStoreApplication) Query(req abcitypes.RequestQuery) abcitypes.ResponseQuery {
+func (app *KVStoreApplication) Query(req abcitypes.RequestQuery) abcitypes.ResponseQuery {
  return abcitypes.ResponseQuery{Code: 0}
 }
 
-func (KVStoreApplication) InitChain(req abcitypes.RequestInitChain) abcitypes.ResponseInitChain {
+func (app *KVStoreApplication) InitChain(req abcitypes.RequestInitChain) abcitypes.ResponseInitChain {
  return abcitypes.ResponseInitChain{}
 }
 
-func (KVStoreApplication) BeginBlock(req abcitypes.RequestBeginBlock) abcitypes.ResponseBeginBlock {
+func (app *KVStoreApplication) BeginBlock(req abcitypes.RequestBeginBlock) abcitypes.ResponseBeginBlock {
  return abcitypes.ResponseBeginBlock{}
 }
 
-func (KVStoreApplication) EndBlock(req abcitypes.RequestEndBlock) abcitypes.ResponseEndBlock {
+func (app *KVStoreApplication) EndBlock(req abcitypes.RequestEndBlock) abcitypes.ResponseEndBlock {
  return abcitypes.ResponseEndBlock{}
 }
 
-func (KVStoreApplication) ListSnapshots(abcitypes.RequestListSnapshots) abcitypes.ResponseListSnapshots {
+func (app *KVStoreApplication) ListSnapshots(abcitypes.RequestListSnapshots) abcitypes.ResponseListSnapshots {
  return abcitypes.ResponseListSnapshots{}
 }
 
-func (KVStoreApplication) OfferSnapshot(abcitypes.RequestOfferSnapshot) abcitypes.ResponseOfferSnapshot {
+func (app *KVStoreApplication) OfferSnapshot(abcitypes.RequestOfferSnapshot) abcitypes.ResponseOfferSnapshot {
  return abcitypes.ResponseOfferSnapshot{}
 }
 
-func (KVStoreApplication) LoadSnapshotChunk(abcitypes.RequestLoadSnapshotChunk) abcitypes.ResponseLoadSnapshotChunk {
+func (app *KVStoreApplication) LoadSnapshotChunk(abcitypes.RequestLoadSnapshotChunk) abcitypes.ResponseLoadSnapshotChunk {
  return abcitypes.ResponseLoadSnapshotChunk{}
 }
 
-func (KVStoreApplication) ApplySnapshotChunk(abcitypes.RequestApplySnapshotChunk) abcitypes.ResponseApplySnapshotChunk {
+func (app *KVStoreApplication) ApplySnapshotChunk(abcitypes.RequestApplySnapshotChunk) abcitypes.ResponseApplySnapshotChunk {
  return abcitypes.ResponseApplySnapshotChunk{}
 }
 ```
 
-Now I will go through each method explaining when it's called and adding
-required business logic.
+### 1.3.1 Add a persistent data store
 
-### 1.3.1 CheckTx
+Our application will need to write its state out to persistent storage so that it
+can stop and start without losing all of its data.
+
+For this tutorial, we will use [badger](https://github.com/dgraph-io/badger). 
+Badger is a fast embedded key-value store. 
+
+First, add badger as a dependency of your go module using the `go get` command:
+
+`go get github.com/dgraph-io/badger`
```

Comment:
The current release is `/v3`. The API changes aren't too substantial, so it might be worth doing that update along with the other changes here. But YMMV.